lib/Auth/Process/PerunAttributes.php

```diff
@@ -81,17 +80,48 @@ public function process(&$request)
         } elseif ($this->mode === self::MODE_PARTIAL) {
             // Check if attribute has some value
             foreach ($this->attrMap as $attrName => $attrValue) {
-                if (isset($request['Attributes'][$attrValue])) {
-                    $attr = $request['Attributes'][$attrValue];
-                    if (empty($attr)) {
+                if (!is_array($attrValue)) {
+                    $attrValue = [$attrValue];
+                }
+                foreach ( $attrValue as $value) {
+                    if (isset($request['Attributes'][$value])) {
+                        $attr = $request['Attributes'][$value];
+                        if (empty($attr)) {
+                            array_push($attributes, $attrName);
+                            break;
+                        }
+                    } else {
                         array_push($attributes, $attrName);
+                        break;
                     }
```

Comment:
I think this could be written in a simpler way. If I understand the code well, you check if the variable "$request['Attributes'][$value]" has some non-empty value. Is there any case where is this "double-check" with isset() and empty() better than using just empty()? I think that using just empty() function should have the same results as this. Correct me if I'm wrong.

If I'm wrong, it could maybe still be simplified by moving these two checks into one if-clause (something like !isset() || (isset() && empty()) ) - maybe a little bit less clear but you wouldn't have to repeat a code in else branch in that case.

Reply:
Yep, seems you are right. The _isset_ is unnecessary and the whole thing can be tested with just _empty_. I have tested it myself, and even if the first key does not exist in the array, the empty will return the result instead of generating an error.

Reply:
Thanks for this point.
I changed it to:
```
                foreach ($attrValue as $value) {
                    if (empty($request['Attributes'][$value])) {
                        array_push($attributes, $attrName);
                        break;
                    }
                }
```